test/test.c

```diff
@@ -515,7 +515,7 @@ static int test_unit(const char *name, bool verbose)
 		for (i=0; i<RE_ARRAY_SIZE(tests); i++) {
 
 			if (verbose) {
-				re_printf("test %u -- %s\n",
+				re_printf("test %zu -- %s\n",
```

Comment:
some of the hunks in this patch are fixing the formatting strings.

it could be an idea to make a separate PR for these fixes ...